Adding example template for API Gateway V2 HTTP integration

tim-littlefair opened the pull request:
This PR adds an extra template under the python3.7 directory which is similar to the existing python3.7/hello_world template but uses the new API Gateway V2 HTTP integration style.

Having spent a few hours trying to work out how to adapt the original REST-style example to change it to use the new style, I felt it would be worth packaging the learning to make life easier for others if the template maintainers are happy to let this go out now.

I apologise for the PR being python3.7 only, my environment presently doesn't default to python3.8 and I 
didn't want to make changes to it in the short term for this purpose.  If you are interested in taking the PR but would require further work to replicate it across other Python versions I am open to spending time making this contribution.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

jfuss commented:
@tim-littlefair I know this PR is old but are you still interested in contributing this PR? 

If so would you be able to do the following:
* Update to remove beta (since it is not in beta anymore)
* Update the template Runtime to just by `python3.7` (examples are per runtime now so no need for the cookiecutter injection).

If you are not interested in or don't have time, no worries. I know it took awhile for someone to review this. We do appreciate the time you spent in contributing.

tim-littlefair commented:
>     * Update to remove beta (since it is not in beta anymore)
> 
>     * Update the template Runtime to just by `python3.7` (examples are per runtime now so no need for the cookiecutter injection).
> 
I've removed references to the API Gateway V2 HTTP integration style being in beta, but as far as I can see the original REST integration style template I cloned this from is still written in cookiecutter format, so I haven't made any changes to that.
As noted in the commit comment, the environment I have now doesn't support Python <= 3.7 so I've updated the runtime to 3.8.
Happy to work  on this further if you point to a diff I should copy to resolve the cookiecutter style.

mndeveci reviewed (changes requested):
Apologies for not reviewing this PR on time after your feedback. 

We have added a new file at the root of this project (`manifest-v2.json`) which keeps the configuration of each template here. Can you also update that file as well?

Thanks again for your contribution!

jfuss commented on python3.8/cookiecutter-aws-sam-hello-http-python/{{cookiecutter.project_name}}/template.yaml:
```diff
@@ -0,0 +1,44 @@
+      {%- elif cookiecutter.runtime == 'python3.8' %}
+      Runtime: python3.8
+      {%- elif cookiecutter.runtime == 'python3.9' %}
+      Runtime: python3.9
```
This example is under the "python3.8" folder so we shouldn't be creating a python3.9 function. @mndeveci Do you want this to be a python3.9 example instead? I can move it or we can leave it as python3.8 for now.

tim-littlefair replied:
I have updated my fork to bring the example into the python3.9 directory, and as suggested I have made python3.9 the default runtime.
Tested with the following commands:
(starting in the example cookiecutter directory)
sam init --location .
cd aws-sam-hello-http-python
sam build --use-container
sam local invoke
sam deploy --guided

NB 'sam deploy --guided' succeeded only if I selected the default region us-east-1.  
If I selected ap-southeast-2 it failed with an error related to the autogenerated S3 bucket containing required artifacts.

hawflau replied:
1. The warning about the reserved logical ID is a known bug in SAM CLI that we haven't addressed yet.
2. I'll try to check if I have the same issue.
Otherwise, the changes look good to me. Thanks for your contribution @tim-littlefair !

hawflau replied:
I could deploy successfully in ap-southeast-2 by running `sam deploy --guided`. I believe the issue you're facing is something unrelated

hawflau reviewed (changes requested):
Could you please also add tests in [tests/integration/build_invoke/test_build_invoke.py](https://github.com/aws/aws-sam-cli-app-templates/blob/master/tests/integration/build_invoke/test_build_invoke.py) and [tests/integration/unit_test/test_unit_test_python3_9.py](https://github.com/aws/aws-sam-cli-app-templates/blob/master/tests/integration/unit_test/test_unit_test_python3_9.py)?

jfuss commented:
Going to close the PR due to no response. Feel free to handle feedback and re-open or create a new PR.
